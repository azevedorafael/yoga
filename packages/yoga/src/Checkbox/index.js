import Switch from './web/Switch';

const Checkbox = {};

Checkbox.Switch = Switch;

export default Checkbox;
