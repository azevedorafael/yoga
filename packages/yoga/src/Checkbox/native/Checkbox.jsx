import React, { useState } from 'react';
import { bool, func, string } from 'prop-types';
import styled from 'styled-components';
import { TouchableWithoutFeedback, View } from 'react-native';
import CheckMark from './CheckMark';

const CheckboxWrapper = styled.View`
  flex-direction: row;
  align-items: center;
`;

const Label = styled.Text(
  ({
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `
  padding-left: ${checkbox.label.padding.left}px;

  font-size: ${checkbox.label.font.size}px;

  color: ${checkbox.label.font.color};
`,
);

const HelperWrapper = styled.View(
  ({
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `
  margin-top: ${checkbox.helper.margin.top}px;
`,
);

const Helper = styled.Text(
  ({
    error,
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `
  font-size: ${checkbox.helper.font.size}px;

  color: ${
    error ? checkbox.helper.selected.font.color : checkbox.helper.font.color
  };
`,
);

const Checkbox = ({
  label,
  helper,
  disabled,
  checked,
  error,
  onPressIn,
  onPressOut,
  ...rest
}) => {
  const [pressed, setPressed] = useState(false);
  return (
    <View>
      <TouchableWithoutFeedback
        {...rest}
        disabled={disabled}
        onPressIn={e => {
          setPressed(true);
          onPressIn(e);
        }}
        onPressOut={e => {
          setPressed(false);
          onPressOut(e);
        }}
      >
        <CheckboxWrapper>
          <CheckMark
            disabled={disabled}
            checked={checked}
            error={error}
            pressed={pressed}
          />
          <Label>{label}</Label>
        </CheckboxWrapper>
      </TouchableWithoutFeedback>
      {helper && (
        <HelperWrapper>
          <Helper error={error}>{helper}</Helper>
        </HelperWrapper>
      )}
    </View>
  );
};

Checkbox.propTypes = {
  label: string,
  /** set a short helper text under checkbox */
  helper: string,
  checked: bool,
  disabled: bool,
  error: bool,
  onPressIn: func,
  onPressOut: func,
};

Checkbox.defaultProps = {
  label: undefined,
  helper: undefined,
  checked: false,
  disabled: false,
  error: false,
  onPressIn: () => {},
  onPressOut: () => {},
};

Checkbox.displayName = 'Checkbox';

export default Checkbox;
