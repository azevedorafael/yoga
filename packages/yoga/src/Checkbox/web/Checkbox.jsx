import React from 'react';
import { bool, string, objectOf, any } from 'prop-types';
import styled from 'styled-components';
import { hexToRgb } from '@gympass/yoga-common';
import { Done } from '@gympass/yoga-icons';

import { HiddenInput } from '../../shared';

const CheckboxWrapper = styled.div`
  display: inline-block;

  * {
    box-sizing: border-box;
  }
`;

const CheckMark = styled.div`
  position: relative;

  border-style: solid;

  ${({
    checked,
    disabled,
    error,
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `
    width: ${checkbox.size}px;
    height: ${checkbox.size}px;

    margin-right: ${checkbox.margin.right}px;

    border-radius: ${checkbox.border.radius}px;
    border-width: ${checkbox.border.width}px;

    border-color: ${
      disabled ? checkbox.disabled.border.color : checkbox.border.color
    };

    ${
      checked
        ? `
        background-color: ${checkbox.checked.backgroundColor}; 

        svg {
          position: absolute;
          top: 50%;
          left: 50%;
          
          fill: ${checkbox.checked.icon.color};

          transform: translate(-50%, -50%);
        }
        `
        : ''
    }

    ${
      disabled && checked
        ? `background-color: ${checkbox.disabled.backgroundColor};`
        : ''
    }

    ${error ? `border-color: ${checkbox.error.border.color};` : ''}

    ${
      error && checked
        ? `background-color: ${checkbox.error.backgroundColor};`
        : ''
    }
  `}
`;

const Label = styled.label`
  display: flex;
  flex-direction: row;
  align-items: center;

  cursor: pointer;

  ${({
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `
    font-size: ${checkbox.label.font.size}px;
    color: ${checkbox.label.font.color};
  `}
`;

const Shadow = styled.span`
  position: absolute;
  display: none;

  ${({
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `
    width: ${checkbox.size}px;
    height: ${checkbox.size}px;
    border-radius: ${checkbox.hover.border.radius}px;
  `}
`;

const CheckboxStyled = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  ${({
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `

  ${Label}:active,
  &:focus-within,
  &:hover  {
    ${Shadow} {
      display: block;
    }
  }

  ${Label}:active {
    ${Shadow} {
      background-color: ${hexToRgb(checkbox.hover.backgroundColor, 0.75)};

      box-shadow: 0 0 0 ${Math.round(
        checkbox.size * 0.33 + checkbox.border.width * 2,
      )}px
      ${hexToRgb(checkbox.hover.backgroundColor, 0.75)};
    }
  }

  &:focus-within {
    ${Shadow} {
      background-color: ${hexToRgb(checkbox.hover.backgroundColor, 0.5)};
      
      box-shadow: 0 0 0 ${Math.round(
        checkbox.size * 0.33 + checkbox.border.width * 2,
      )}px
        ${hexToRgb(checkbox.hover.backgroundColor, 0.5)};
      }
  }

  &:hover {
    ${Shadow} {
      background-color: ${hexToRgb(checkbox.hover.backgroundColor, 0.25)};

      box-shadow: 0 0 0 ${Math.round(
        checkbox.size * 0.33 + checkbox.border.width * 2,
      )}px
      ${hexToRgb(checkbox.hover.backgroundColor, 0.25)};
    }
  }
  `}
`;

const HelperWrapper = styled.div`
  width: 100%;

  ${({
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `
    margin-top: ${checkbox.helper.margin.top}px;
  `}
`;

const Helper = styled.span`
  ${({
    error,
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `
    color: ${
      error ? checkbox.helper.selected.font.color : checkbox.helper.font.color
    };

    font-size: ${checkbox.helper.font.size}px;
  `}
`;

const Checkbox = ({
  value,
  label,
  helper,
  disabled,
  checked,
  error,
  style,
  className,
  ...rest
}) => (
  <CheckboxWrapper style={style} className={className}>
    <CheckboxStyled>
      <Label>
        <Shadow />
        <CheckMark disabled={disabled} checked={checked} error={error}>
          {checked && <Done />}
        </CheckMark>
        <HiddenInput
          type="checkbox"
          checked={checked}
          disabled={disabled}
          {...rest}
        />
        {label}
      </Label>
    </CheckboxStyled>
    {helper && (
      <HelperWrapper>
        <Helper error={error}>{helper}</Helper>
      </HelperWrapper>
    )}
  </CheckboxWrapper>
);

Checkbox.propTypes = {
  label: string,
  /** short helper text under checkbox */
  helper: string,
  value: string,
  checked: bool,
  disabled: bool,
  error: bool,
  /** set a style to the checkbox container */
  style: objectOf(any),
  className: string,
};

Checkbox.defaultProps = {
  label: undefined,
  value: undefined,
  helper: undefined,
  checked: false,
  disabled: false,
  error: false,
  style: undefined,
  className: undefined,
};

Checkbox.displayName = 'Checkbox';

export default Checkbox;
