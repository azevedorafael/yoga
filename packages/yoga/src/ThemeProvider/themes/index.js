import corp from './corp';
import endUser from './end-user';
import gyms from './gyms';

export { endUser as default, endUser, corp, gyms };
