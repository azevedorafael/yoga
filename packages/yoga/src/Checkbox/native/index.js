import Switch from './Switch';
import Checkbox from './Checkbox';

export { Checkbox, Switch };
