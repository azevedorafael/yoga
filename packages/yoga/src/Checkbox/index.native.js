import { Checkbox, Switch } from './native';

Checkbox.Switch = Switch;

export default Checkbox;
