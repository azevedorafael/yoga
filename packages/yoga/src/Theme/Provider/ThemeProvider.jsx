import React from 'react';
import { ThemeProvider as SCThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';
import * as tokens from '@gympass/yoga-tokens';
import * as themes from '../themes';

const getTheme = ({ theme, locale }) => {
  const token = tokens[locale] || tokens.default;
  const appTheme = themes[theme] || themes.default;

  return appTheme(token);
};

const ThemeProvider = ({ children, ...theme }) => (
  <SCThemeProvider theme={{ yoga: getTheme(theme) }}>
    {children}
  </SCThemeProvider>
);

ThemeProvider.propTypes = {
  theme: PropTypes.string,
  locale: PropTypes.string,
  children: PropTypes.node.isRequired,
};

ThemeProvider.defaultProps = {
  theme: 'endUser',
  locale: 'pt-BR',
};

export default ThemeProvider;
