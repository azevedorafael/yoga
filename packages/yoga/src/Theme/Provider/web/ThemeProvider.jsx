import React from 'react';
import { node } from 'prop-types';

import Provider from '../Provider';
import GlobalStyle from './GlobalStyle';

/** This component provides a theme to all React components underneath itself via the context API. */
const ThemeProvider = ({ children, ...rest }) => (
  <Provider {...rest}>
    <GlobalStyle />
    {children}
  </Provider>
);

ThemeProvider.propTypes = {
  children: node.isRequired,
};

export default ThemeProvider;
