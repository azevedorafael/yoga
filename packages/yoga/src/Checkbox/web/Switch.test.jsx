import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Checkbox from '..';
import ThemeProvider from '../../Theme';

describe('<Checkbox />', () => {
  describe('<Checkbox.Switch />', () => {
    describe('Snapshots', () => {
      it('should match snapshot with default switch', () => {
        const { container } = render(
          <ThemeProvider>
            <Checkbox.Switch />
          </ThemeProvider>,
        );
        expect(container).toMatchSnapshot();
      });

      it('should match snapshot with disabled switch', () => {
        const { container } = render(
          <ThemeProvider>
            <Checkbox.Switch disabled />
          </ThemeProvider>,
        );
        expect(container).toMatchSnapshot();
      });

      it('should match snapshot with checked switch', () => {
        const { container } = render(
          <ThemeProvider>
            <Checkbox.Switch checked />
          </ThemeProvider>,
        );
        expect(container).toMatchSnapshot();
      });

      it('should match snapshot with disabled and checked switch', () => {
        const { container } = render(
          <ThemeProvider>
            <Checkbox.Switch disabled checked />
          </ThemeProvider>,
        );
        expect(container).toMatchSnapshot();
      });

      it('should call onChange function when click on Checkbox.Switch', () => {
        const onChangeMock = jest.fn();
        const { getByRole } = render(
          <ThemeProvider>
            <Checkbox.Switch disabled checked onChange={onChangeMock} />
          </ThemeProvider>,
        );

        fireEvent.click(getByRole('switch', { hidden: true }));

        expect(onChangeMock).toHaveBeenCalled();
      });
    });
  });
});
