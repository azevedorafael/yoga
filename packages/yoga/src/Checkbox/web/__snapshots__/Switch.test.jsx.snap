// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> <Checkbox.Switch /> Snapshots should match snapshot with checked switch 1`] = `
.c2 {
  position: absolute;
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: inherit;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: pointer;
  width: 48px;
  height: 24px;
  background-color: #F46152;
  border-radius: 9999px;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
}

.c0 * {
  box-sizing: border-box;
}

.c1:focus + .c4,
.c1:hover + .c4 {
  box-shadow: 0 0 0 8px rgba(152,152,166,0.2);
}

.c3 {
  position: relative;
  pointer-events: none;
  left: 4px;
  width: 16px;
  height: 16px;
  background-color: #FFFFFF;
  border-radius: 9999px;
  -webkit-transition-property: left,-webkit-transform;
  -webkit-transition-property: left,transform;
  transition-property: left,transform;
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
  left: calc(100% - 4px);
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

.c1:focus + .c3,
.c1:hover + .c3 {
  box-shadow: 0 0 0 8px rgba(244,97,82,0.2);
}

<div>
  <label
    class="c0"
  >
    <input
      aria-checked="true"
      aria-hidden="true"
      aria-readonly="false"
      class="c1 c2"
      role="switch"
      type="checkbox"
    />
    <span
      class="c3"
      role="button"
    />
  </label>
</div>
`;

exports[`<Checkbox /> <Checkbox.Switch /> Snapshots should match snapshot with default switch 1`] = `
.c2 {
  position: absolute;
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: inherit;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: pointer;
  width: 48px;
  height: 24px;
  background-color: #DBDBE8;
  border-radius: 9999px;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
}

.c0 * {
  box-sizing: border-box;
}

.c3 {
  position: relative;
  pointer-events: none;
  left: 4px;
  width: 16px;
  height: 16px;
  background-color: #FFFFFF;
  border-radius: 9999px;
  -webkit-transition-property: left,-webkit-transform;
  -webkit-transition-property: left,transform;
  transition-property: left,transform;
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
}

.c1:focus + .c3,
.c1:hover + .c3 {
  box-shadow: 0 0 0 8px rgba(152,152,166,0.2);
}

<div>
  <label
    class="c0"
  >
    <input
      aria-checked="false"
      aria-hidden="true"
      aria-readonly="false"
      class="c1 c2"
      role="switch"
      type="checkbox"
    />
    <span
      class="c3"
      role="button"
    />
  </label>
</div>
`;

exports[`<Checkbox /> <Checkbox.Switch /> Snapshots should match snapshot with disabled and checked switch 1`] = `
.c2 {
  position: absolute;
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: inherit;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: pointer;
  width: 48px;
  height: 24px;
  background-color: #F46152;
  border-radius: 9999px;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
  background-color: #F5F5FA;
  cursor: not-allowed;
}

.c0 * {
  box-sizing: border-box;
}

.c1:focus + .c4,
.c1:hover + .c4 {
  box-shadow: 0 0 0 8px rgba(152,152,166,0.2);
}

.c1:focus + .c5,
.c1:hover + .c5 {
  box-shadow: 0 0 0 8px rgba(244,97,82,0.2);
}

.c3 {
  position: relative;
  pointer-events: none;
  left: 4px;
  width: 16px;
  height: 16px;
  background-color: #FFFFFF;
  border-radius: 9999px;
  -webkit-transition-property: left,-webkit-transform;
  -webkit-transition-property: left,transform;
  transition-property: left,transform;
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
  left: calc(100% - 4px);
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  background-color: #DBDBE8;
  cursor: not-allowed;
}

<div>
  <label
    class="c0"
    disabled=""
  >
    <input
      aria-checked="true"
      aria-hidden="true"
      aria-readonly="true"
      class="c1 c2"
      disabled=""
      role="switch"
      type="checkbox"
    />
    <span
      class="c3"
      disabled=""
      role="button"
    />
  </label>
</div>
`;

exports[`<Checkbox /> <Checkbox.Switch /> Snapshots should match snapshot with disabled switch 1`] = `
.c2 {
  position: absolute;
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: inherit;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: pointer;
  width: 48px;
  height: 24px;
  background-color: #DBDBE8;
  border-radius: 9999px;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
  background-color: #F5F5FA;
  cursor: not-allowed;
}

.c0 * {
  box-sizing: border-box;
}

.c1:focus + .c4,
.c1:hover + .c4 {
  box-shadow: 0 0 0 8px rgba(152,152,166,0.2);
}

.c3 {
  position: relative;
  pointer-events: none;
  left: 4px;
  width: 16px;
  height: 16px;
  background-color: #FFFFFF;
  border-radius: 9999px;
  -webkit-transition-property: left,-webkit-transform;
  -webkit-transition-property: left,transform;
  transition-property: left,transform;
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
  background-color: #DBDBE8;
  cursor: not-allowed;
}

<div>
  <label
    class="c0"
    disabled=""
  >
    <input
      aria-checked="false"
      aria-hidden="true"
      aria-readonly="true"
      class="c1 c2"
      disabled=""
      role="switch"
      type="checkbox"
    />
    <span
      class="c3"
      disabled=""
      role="button"
    />
  </label>
</div>
`;
